Integration of ZDC fastsim prototype into O2.

SwirtaB opened the pull request:
Prototype provides interface for running two ML models in O2. CPU only for now.

Had to revert commit dd8e8f134041d3b8c5875042a8c9bf1266c39a14 due to compile error.

Also it is required to add _ONNXRuntime_ entry in o2.sh in alidist.

sawenzel commented on Utilities/ShmManager/src/ShmManager.cxx:
```diff
@@ -75,7 +75,6 @@ struct ShmManager {
       }
       uint16_t id = stoi(conf.at(0));
       uint64_t size = stoull(conf.at(1));
-      segmentCfgs.emplace_back(fair::mq::shmem::SegmentConfig{id, size, "rbtree_best_fit"});
```
Please don't change this file. It has nothing to do with your feature and just taking it out is not a solution.
If you have problem with this file, please open a JIRA ticket describing the issue.

SwirtaB replied:
Sorry for that. I revoked those changes in newest commits.

sawenzel reviewed (dismissed):
Approving this to trigger CI testing once. Please don't merge yet.

sawenzel commented on Detectors/ZDC/fastsimulation/include/FastSimulations.hpp:
```diff
@@ -0,0 +1,99 @@
+class SAEModelSimulation : public NeuralFastSimulation
+{
+ public:
+  SAEModelSimulation(std::array<float, 9> conditionalMeans, std::array<float, 9> conditionalScales);
```
try to use `const &` consistently in all interfaces taking complex objects

SwirtaB replied:
Fixed in newest commits.

sawenzel commented:
Thanks. Let's see how it compiles. 

A couple of preliminary comments:
a) please add copyright notices to all new files
b) please format your code with git-clang-format
c) please follow conventions for filenames and suffices (we typically use .h instead of .hpp and .cxx for source)
d) please follow conventions for variable names (member variables are `mFooBar`) and function names (the main Readme contains links to coding conventions)
e) please don't commit binary files. They need to be distributed by other means (we have CVMFS for this)
f) please put the stuff into `o2::` namespace (better `o2::zdc::fastsim`)

SwirtaB committed:
Added fastsim build requirements

Added find_package(ONNXRuntime) to O2 CMakeLists
Added add_subdirectory(fastsim) to Detectors/ZDC CMakeLists

SwirtaB committed:
Fixed fastsim integration

Fixed code and file naming
Removed binary files

SwirtaB committed:
Added gurads to CMakeLists

Now ONNXRuntime is optional and if not present
fastsim module would not build

SwirtaB commented:
Added conditional build in CMakeLists.txt. ONNXRuntime is no longer required to build O2 (fast simulation module won't build).

sawenzel committed:
Update CMakeLists.txt

sawenzel commented:
@SwirtaB : I think we are getting there. The only remaining issue is to fix the full-CI test. It complains with:
```
## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:46:13: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:139:23: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:143:25: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:147:26: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:209:23: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:213:25: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/FastSimulations.cxx:217:26: error: statement should be inside braces [readability-braces-around-statements]
/sw/SOURCES/O2/8162-slc8_x86-64/0/Detectors/ZDC/fastsimulation/src/Utils.cxx:50:22: error: statement should be inside braces [readability-braces-around-statements]
--
```

SwirtaB committed:
Fixed O2 CodeChecker errors

Added missing braces to fix error: readability-braces-around-statements

SwirtaB committed:
Merge branch 'dev' of https://github.com/SwirtaB/AliceO2 into dev

SwirtaB committed:
Fix typo

SwirtaB commented:
I have fixed those errors, should be fine now.
